### PR TITLE
V3/improve tests

### PR DESCRIPTION
**Pull request by josepot**
There is still a lot of room for improvement, but now we've got 100% of test coverage, which I think it's pretty good.

Also, while I was making some of these tests I noticed a couple of things that had to be improved ([1](https://github.com/josepot/rereducer/commit/621b3f4f6b4b1f62375474fcb5faf93139a3e798), [2](https://github.com/josepot/rereducer/commit/6afc331f90439c273b129d001e8c251771e592e6)). Which is noice!

Well, there is just 1 branch that I have not been able to test ([this one](https://github.com/josepot/rereducer/blob/v3.0.0-alpha.1/src/switchReducers.js#L46)) and I'm not going to go all nuts trying to get it tested 😄

### CODE REVIEW

File: test/createReducer.spec.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { createReducer, fromAction, getState } from '../src'
+
+describe('createReducer', () => {
+  const dependencies = ['foo', 'bar', 'baz'].map(fromAction).concat(getState)
+  const state = 'test'
+  const [foo, bar, baz] = ['fooVal', 'barVal', 'bazVal']
+
+  it(`Accepts a list of branching functions and converging function and returns a new function. When invoked, this new function is applied to some arguments, each branching function is applied to those same arguments. The results of each branching function are passed as arguments to the converging function to produce the return value.`, () => {
```
Comment (voliva) [inline test/createReducer.spec.js]: Is there any way to break the description of this test down or make it simpler to read?

Comment (josepot) [reply test/createReducer.spec.js]: I'm open to suggestions :smile:

Comment (voliva) [reply test/createReducer.spec.js]: Yes... that's quite hard. I think you can completely skip the first sentence, and try to make the rest smaller. This is the best I could think of:

```
it('creates a function that will return computeFn called with the result of executing each of the dependencies with the original arguments')
```

Still not 100% happy with this.

Comment (josepot) [reply test/createReducer.spec.js]: Ok, what about this?
```suggestion
  it(`receives 2 parameters: an Array of dependency functions and a compute function. It returns a function that invokes the compute function using the results of the dependencies as the arguments`, () => {
```

**Comment by voliva** [comment]
Ah yes, that's perfect :+1:

**Review by voliva — APPROVED** [review]
Sorry I forgot to approve :sweat_smile:

**Pull request merged**